### PR TITLE
Automorphism groups for polytopes

### PR DESCRIPTION
**Pull request by lkastner**

### CODE REVIEW

File: src/PolyhedralGeometry/Groups.jl [hunk 1]
```diff
@@ -164,7 +164,16 @@ end
 @doc Markdown.doc"""
     automorphism_group(P::Polyhedron; type = :combinatorial)
```
Comment (thofma) [inline src/PolyhedralGeometry/Groups.jl]: How about an `:action` keyword argument which could be `:vertex` or `:facet`? If this makes no sense or does not work out in practice, just discard my comment.

Comment (lkastner) [reply src/PolyhedralGeometry/Groups.jl]: Let me know whether I understood you right.

**Comment by lkastner** [comment]
FYI: This PR is waiting for
https://github.com/oscar-system/Polymake.jl/pull/404
to get merged.

File: src/PolyhedralGeometry/Groups.jl (hunk 1 above)
Comment (thofma) [reply src/PolyhedralGeometry/Groups.jl]: Yes, this is what I meant.

File: src/PolyhedralGeometry/Groups.jl [hunk 2]
```diff
@@ -112,9 +109,218 @@ julia> length(elements(G))
...
+  :on_facets   => [(1,2)(3,4), (1,3)]
+
+julia> automorphism_group_generators(quad; type = :linear)
+Dict{Symbol, Vector{PermGroupElem}} with 2 entries:
+  :on_vertices => [(2,4)]
+  :on_facets   => [(1,2)(3,4)]
+```
+"""
+function automorphism_group_generators(P::Polyhedron; type = :combinatorial, action = :all)
+    pm_object(P).BOUNDED || throw(ArgumentError("Automorphism groups not supported for unbounded polyhedra."))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Groups.jl]:
```suggestion
    is_bounded(P) || throw(ArgumentError("Automorphism groups not supported for unbounded polyhedra."))
```

File: src/PolyhedralGeometry/Groups.jl [hunk 3]
```diff
@@ -112,9 +109,218 @@ julia> length(elements(G))
...
+                :on_cols => cols_action)
+    else
+        throw(ArgumentError("Action $(action) not supported."))
+    end
+end
+    
+
+
+function _linear_symmetries_generators(P::Polyhedron; action = :all)
     if pm_object(P).BOUNDED
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Groups.jl]:
```suggestion
    if is_bounded(P)
```

File: src/PolyhedralGeometry/Groups.jl [hunk 4]
```diff
@@ -70,11 +70,8 @@ julia> length(elements(G))
 ```
 """
 function combinatorial_symmetries(P::Polyhedron)
-    if pm_object(P).BOUNDED
-        pm_group_to_oscar_group(Polymake.polytope.combinatorial_symmetries(pm_object(P)))
-    else
-        throw(ArgumentError("Symmetry groups currently supported for bounded polyhedra only"))
-    end
+    result = automorphism_group(P; type=:combinatorial)
+    return result[:on_vertices]
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Groups.jl]:
```suggestion
    return automorphism_group(P; type=:combinatorial, action=:on_vertices)
```

File: src/Combinatorics/Graphs.jl [hunk 5]
```diff
@@ -542,16 +542,34 @@ Return generators of the automorphism group of the graph `g`.
...
+ (3,4)
+ (2,3)
+ (1,2)
 ```
 """
 function automorphism_group_generators(g::Graph{T}) where {T <: Union{Directed, Undirected}}
     pmg = pm_object(g);
     result = Polymake.graph.automorphisms(pmg)
-    return [[x+1 for x in a] for a in result]
+    return pm_arr_arr_to_group_generators(result)
```
Comment (benlorenz) [inline src/Combinatorics/Graphs.jl]:
```suggestion
    return _pm_arr_arr_to_group_generators(result)
```

File: src/Combinatorics/Graphs.jl [hunk 6]
```diff
@@ -542,16 +542,34 @@ Return generators of the automorphism group of the graph `g`.
...
+# Examples
+```jldoctest
+julia> g = complete_graph(4);
+
+julia> automorphism_group(g)
+Group([ (3,4), (2,3), (1,2) ])
+```
+"""
+function automorphism_group(g::Graph{T}) where {T <: Union{Directed, Undirected}}
+    return gens_to_group(automorphism_group_generators(g))
```
Comment (benlorenz) [inline src/Combinatorics/Graphs.jl]:
```suggestion
    return _gens_to_group(automorphism_group_generators(g))
```

**Review by benlorenz — APPROVED** [review]

**Pull request merged**